gt/main.c

```diff
@@ -178,9 +375,26 @@ init_gt_instances(struct gt_config *gt_conf)
 			goto out;
 		}
 		inst_ptr->tx_queue = ret;
+
+		/*
+		 * Set up the lua state for each instance,
+		 * and initialize the policy tables.
+		 */
+		ret = config_gt_instance(gt_conf, lcore);
+		if (ret < 0)
+			goto free_lua_state;
+
+		num_succ_instances++;
 	}
 
 	ret = 0;
+	goto out;
+
+free_lua_state:
+	lua_close(inst_ptr->lua_state);
+	inst_ptr->lua_state = NULL;
+	for (i = 0; i < num_succ_instances; i++)
+		lua_close(gt_conf->instances[i].lua_state);
```

Comment:
There should be a function `cleanup_gt_instance()` here. If all `cleanup_gt_instance()` does is to call `lua_close()`, fine, make inline. But calling `lua_close()` here instead of `cleanup_gt_instance()` might introduce a bug later when someone needs to write `cleanup_gt_instance()` because there're other fields to clean up, and forgets to change it here as well.